### ISSUE TITLE
Deprecate or remove use __file__

### ISSUE THREAD

**Issue by jayvdb** [john-kurkowski/tldextract#196]
For reading `.tld_set`, `__file__` should be replaced with pkgutil.get_data, which is used for `.tld_set_snapshot`.

The dataset file may have been placed into the package data via another means, and the attribute `__file__` may not be present at all, as it is an optional attribute of the Python module loading system.  tldextract fails badly atm when it is missing.  An example of when it is missing is PyOxidizer, and Nuitka also has similar problems.

When it is missing, or when the package directory is not writable, the cache should be stored in a different location without a warning.

Currently I see the following for every invocation which imports tldextract, because it is installed as a system package used by system apps:

> WARNING unable to cache TLDs in file /usr/lib/python3.8/site-packages/tldextract/.tld_set: [Errno 13] Permission denied: '/usr/lib/python3.8/site-packages/tldextract/.tld_set'

**Comment by john-kurkowski** [comment]
Makes sense. I don't suppose it's as easy as this? Plus changing types from filename strings to read-in-file strings.

```diff
-CACHE_FILE_DEFAULT = os.path.join(os.path.dirname(__file__), '.tld_set')
+CACHE_FILE_DEFAULT = pkgutil.get_data(__name__, '.tld_set')
```

**Comment by john-kurkowski** [comment]
> When it is missing, or when the package directory is not writable, the cache should be stored in a different location without a warning.

Is there a standard, usually writable location for this? Or a library to determine it?

**Comment by jayvdb** [comment]
The xdg standard defines cache directories.

[`appdirs`](https://pypi.org/project/appdirs/) package is typical solution.

**Comment by john-kurkowski** [comment]
That is what I've been looking for. Maybe that's the most robust solution to #9, #149, and this ticket.

**Comment by merwok** [comment]
FYI, `platformdirs` is a replacement for `appdirs` that’s actively maintained and more performant.
